sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala

```diff
@@ -128,7 +128,9 @@ class FileBasedDataSourceSuite extends QueryTest
   }
 
   allFileBasedDataSources.foreach { format =>
-    test(s"SPARK-23372 error while writing empty schema files using $format") {
+    val emptySchemaValidationConf = SQLConf.ALLOW_EMPTY_SCHEMAS_FOR_WRITES.key
+    test(s"SPARK-23372 error while writing empty schema files " +
+      s"using $format and ${emptySchemaValidationConf} disabled") {
```

Comment:
Maybe we should leave this test name as-is? Since disabled is the default. It makes this code much more messy.